src/client.rs

```diff
@@ -142,6 +144,7 @@ impl Client {
             server_info: Arc::new(Mutex::new(ServerInfo::default())),
             shutdown: Arc::new(Mutex::new(false)),
             options: Arc::new(options),
+            join_handles: Arc::new(Mutex::new(Vec::new())),
```

Comment:
Sure we don't want to just name these? should be two handles iirc (looked into this on Friday).

Reply:
agree, changed.